Source/ui_unix/mainwindow.cpp

```diff
@@ -448,3 +451,27 @@ void MainWindow::on_actionController_Manager_triggered()
     ControllerConfigDialog ccd;
     ccd.exec();
 }
+
+void MainWindow::on_actionCapture_Screen_triggered()
+{
+    static_cast<CGSH_OpenGL*>(g_virtualMachine->m_ee->m_gs)->OnFlipImplComplete.connect_extended(
+        [&](const boost::signals2::connection &c)->void
+        {
+            c.disconnect();
+            try
+            {
+                auto m_buffer = static_cast<CGSH_OpenGL*>(g_virtualMachine->m_ee->m_gs)->GetScreenshot();
+                auto name = CAppConfig::GenerateScreenShotPath(g_virtualMachine->m_ee->m_os->GetExecutableName());
+                Framework::CStdStream outputStream(name.string().c_str(), "wb");
+                Framework::CBMP::WriteBitmap(m_buffer, outputStream);
+            }
+            catch (const std::exception& e)
+            {
+                QMessageBox messageBox;
```

Comment:
Instead of displaying a message box, I think it would be more interesting to write a message in the status bar.